test/azure/availabilityset_test.go

```diff
@@ -0,0 +1,74 @@
+// +build azure
+
+// NOTE: We use build tags to differentiate azure testing because we currently do not have azure access setup for
+// CircleCI.
+
+package test
+
+import (
+	"testing"
+
+	"github.com/gruntwork-io/terratest/modules/azure"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	// subscriptionID is overridden by the environment variable "ARM_SUBSCRIPTION_ID"
+	subscriptionID = ""
+	rgName         = "terratest-rg"
+)
+
+/*
+The below tests are currently stubbed out, with the expectation that they will throw errors.
+If/when methods to create and delete network resources are added, these tests can be extended.
+*/
+
+func TestGetAvailabilitySetClientE(t *testing.T) {
```

Comment:
It seems like the way to test these is to deploy the example you created above using Terratest's `terraform` package and then use these methods to validate that what got deployed is working...

Reply:
Agreed, just stubbed out the methods to confirm consistency and signatures as noted in the file comments. I can remove if preferred.

New location: modules/azure/availabilityset_test.go